### PR TITLE
Use sqlite

### PR DESCRIPTION
**Pull request by JakubFornadel**

### CODE REVIEW

File: smart_contracts/db_resource.cpp [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#include "sophiatx/smart_contracts/db_resource.h"
```
Comment (ejossev) [inline smart_contracts/db_resource.cpp]: Can we maintain the convention of naming c++ headers with "hpp" suffix?

File: smart_contracts/include/sophiatx/smart_contracts/db_resource.h [hunk 2]
```diff
@@ -0,0 +1,33 @@
+#ifndef SOPHIATX_DB_RESOURCE_H
+#define SOPHIATX_DB_RESOURCE_H
```
Comment (ejossev) [inline smart_contracts/include/sophiatx/smart_contracts/db_resource.h]: Why not use #pragma once instead?

**Review by ejossev — COMMENTED** [review]
Can we also have a short (5liner) example how this is used, especially the "bind" methods in conjunction with (mutable_)variant(_object)? I'm lost especially with the `?`, `:`, `@` and `$` notations...

**Review by MatusKysel — APPROVED** [review]

File: smart_contracts/db_resource.cpp (hunk 1 above)
Comment (JakubFornadel) [reply smart_contracts/db_resource.cpp]: yes, my bad :)

File: smart_contracts/include/sophiatx/smart_contracts/db_resource.h (hunk 2 above)
Comment (JakubFornadel) [reply smart_contracts/include/sophiatx/smart_contracts/db_resource.h]: "even though #pragma once is widely supported, it is not part of standard." And i kind of like GUARDS more anyway... I dont think there is much difference

**Pull request merged**